### PR TITLE
math, date, text argumant validation, fix ceiling, exact, floor, arab…

### PR DESCRIPTION
**Pull request by jhkcia**
- implement argument checking, return an error or an invalid number of arguments. validations are based on official ms excel documentation on functions.
- fix Arabic function to work with lower case strings, now it parses both IX, ix.
- fix ceiling and ceiling.math signature, [[CEILING](https://support.microsoft.com/en-us/office/ceiling-function-0a5cd7c8-0720-4f0a-bd2c-c943e510899f)]
- some math functions do not exist in excel, I left some comments on them. [E, GT, GTE, LT, LTE, EQ, NE, POW, add, minus, divide, multiply], maybe it is better to remove them.
- some text functions do not exist in excel, [REGEXEXTRACT, REGEXMATCH, REGEXREPLACE, SPLIT], may be better to remove them.
- exact function always returned true, Fixed it.
- fix floor function

### CODE REVIEW

File: lib/date-time.js [hunk 1]
```diff
@@ -64,6 +64,9 @@ var WEEKEND_TYPES = [
 ];
 
 exports.DATE = function (year, month, day) {
+  if (arguments.length != 3) {
```
Comment (dam1r89) [inline lib/date-time.js:67]: Could you please use strict equality everywhere

File: lib/text.js [hunk 2]
```diff
@@ -45,6 +54,9 @@ exports.CODE = function(text) {
 };
 
 exports.CONCATENATE = function() {
+  if (!arguments || arguments.length === 0) {
```
Comment (dam1r89) [inline lib/text.js:57]: Is it possible not to have arguments?

**Comment by nicolashefti** [comment]
@jhkcia can you remove your comments in the PR and open an issue to discuss theses points:
- some math functions do not exist in excel, I left some comments on them. [E, GT, GTE, LT, LTE, EQ, NE, POW, add, minus, divide, multiply], maybe it is better to remove them.
- some text functions do not exist in excel, [REGEXEXTRACT, REGEXMATCH, REGEXREPLACE, SPLIT], may be better to remove them.

**Comment by nicolashefti** [comment]
@pbadenski you made a lot of work (#53 #64) in terms of parity between Excel and Formula.js for undefined/error cases. This PR goes further and also modifies some of the error types you proposed. What do you think?

**Comment by nicolashefti** [comment]
@pbadenski any thoughts on this?

**Comment by pbadenski** [comment]
Sorry - I'm reviewing this today and will come back soon.

**Comment by pbadenski** [comment]
General comments:
 - I believe that if checks on argument number conformance should return "errors.error" - This imho most closely matches Excel behaviour - as Excel would not even allow you to define function without required arguments (displays modal dialog and won't save the edit until corrected). We probably weren't consistent before on what was returned for these checks. With this number of new checks introduced I think we should standardise on "errors.error". 
 - Pull request mentions "fix floor function" - I couldn't find any specific fixes to the floor function
 - ditto to @nicolashefti comment: https://github.com/formulajs/formulajs/pull/78#issuecomment-981142920

File: lib/text.js [hunk 3]
```diff
@@ -172,11 +187,17 @@ exports.LOWER = function(text) {
 };
 
 exports.MID = function(text, start, number) {
+  if(arguments.length!=3){
```
Comment (pbadenski) [inline lib/text.js:190]: please change to strict equality

File: lib/statistical.js [hunk 4]
```diff
@@ -1879,6 +2143,9 @@ exports.VARPA = function() {
 exports.WEIBULL = {};
 
 exports.WEIBULL.DIST = function(x, alpha, beta, cumulative) {
+  if( arguments.length!=4){
```
Comment (pbadenski) [inline lib/statistical.js:2146]: please change to strict equality

**Comment by nicolashefti** [comment]
No more activity on this PR.

**Pull request closed without merging**